internal/pkg/service/stream/storage/node/readernode/sliceupload/operator_test.go

```diff
@@ -0,0 +1,371 @@
+		Size:     5 * datasize.MB,
+		Interval: duration.From(5 * time.Minute),
+	}
+
+	// The interval triggers upload conditions check
+	conditionsCheckInterval := time.Second
+
+	nodeID := "my-volume"
+	volumesPath := t.TempDir()
+	volumePath1 := filepath.Join(volumesPath, "hdd", "001")
+	require.NoError(t, os.MkdirAll(volumePath1, 0o700))
+	require.NoError(t, os.WriteFile(filepath.Join(volumePath1, volume.IDFile), []byte(fmt.Sprintf("%s-1", nodeID)), 0o600))
+	volumePath2 := filepath.Join(volumesPath, "hdd", "002")
+	require.NoError(t, os.MkdirAll(volumePath2, 0o700))
+	require.NoError(t, os.WriteFile(filepath.Join(volumePath2, volume.IDFile), []byte(fmt.Sprintf("%s-2", nodeID)), 0o600))
+	ignoredEtcdKeys := etcdhelper.WithIgnoredKeyPattern("^(definition)|(runtime)|(storage/file)|(storage/slice)|(storage/stats)|(storage/volume)")
+
+	// Create dependencies
+	clk := clock.NewMock()
+	clk.Set(utctime.MustParse("2000-01-01T00:00:00.000Z").Time())
+	deps, mock := dependencies.NewMockedCoordinatorScopeWithConfig(t, func(cfg *config.Config) {
+		cfg.NodeID = nodeID
+		cfg.Hostname = "localhost"
+		cfg.Storage.VolumesPath = volumesPath
+		cfg.Storage.Level.Staging.Upload.Trigger = uploadTrigger
+		cfg.Storage.Level.Staging.Operator.CheckInterval = duration.From(conditionsCheckInterval)
+	}, commonDeps.WithClock(clk))
+	logger := mock.DebugLogger()
+	client := mock.TestEtcdClient()
+
+	// Create dependencies
+	d, err := dependencies.NewStorageReaderScope(ctx, deps, mock.TestConfig())
+	require.NoError(t, err)
+	/*d, mock := dependencies.NewMockedReaderNodeScopeWithConfig(t, func(cfg *config.Config) {
+		//cfg.Storage.Level.Target.Import.Trigger = importTrigger
+		cfg.Storage.Level.Target.Operator.CheckInterval = duration.From(conditionsCheckInterval)
+		//commonDeps.WithEtcdConfig(etcdCfg),
+	}, commonDeps.WithClock(clk))*/
+
+	// Block switching to the uploading state, a source is using slice
+	sourceNode, err := closesync.NewSourceNode(d, "source-node")
+	require.NoError(t, err)
+
+	// Start slice rotation coordinator
+	require.NoError(t, slicerotation.Start(deps, mock.TestConfig().Storage.Level.Staging.Operator))
+
+	// Start slice upload reader node
+	require.NoError(t, sliceupload.Start(d, mock.TestConfig().Storage.Level.Staging.Operator))
+
+	// apiCtx - for operations triggered by an authorized API call
+	apiCtx := rollback.ContextWith(ctx, rollback.New(d.Logger()))
+	apiCtx = context.WithValue(apiCtx, dependencies.KeboolaProjectAPICtxKey, mock.KeboolaProjectAPI())
+
+	// Register mocked responses
+	// -----------------------------------------------------------------------------------------------------------------
+	url := bridgeTest.GCSServer(t)
+	os.Setenv("STORAGE_EMULATOR_HOST", url)
+	transport := mock.MockedHTTPTransport()
+	{
+		bridgeTest.MockTokenStorageAPICalls(t, transport)
+		bridgeTest.MockBucketStorageAPICalls(t, transport)
+		bridgeTest.MockTableStorageAPICalls(t, transport)
+		bridgeTest.MockGCSBucket(t, transport)
```

Comment:
New Mocked GCS Bucket responses.

Reply:
Here shouldn't be no Keboola specific test.

Please:
- Use a `dummy` sink in the `sliceupload` test.
- Move Keboola specific tests to the `keboola/bridge` package.
- You don't need to mock both, S3 and GCS, go-client has tests for them (maybe tests with a real test project will be better).